Add EARN-E P1 Meter integration documentation (#43824)
home-assistant/home-assistant.github.io: source/_integrations/earn_e_p1.markdown

---
title: EARN-E P1 Meter
description: Monitor your smart meter's energy and gas data in real-time with the EARN-E P1 Meter integration.
ha_category:
  - Energy
  - Sensor

> ha_release: 2026.4

Reviewer: ```suggestion
ha_release: 2026.5
```

Author: Already at `2026.5` on the branch.

### Meter reading sensors (~60 second updates)

- **Energy delivered tariff 1**: Total energy consumed on tariff 1 (kWh)
- **Energy delivered tariff 2**: Total energy consumed on tariff 2 (kWh)
- **Energy returned tariff 1**: Total energy returned on tariff 1 (kWh)
- **Energy returned tariff 2**: Total energy returned on tariff 2 (kWh)
- **Gas delivered**: Total gas consumed (m³)
- **Wi-Fi RSSI**: Wi-Fi signal strength of the device (dBm)

## Prerequisites

The EARN-E energy monitor must be:

- Connected to your smart meter's P1 port
- Connected to the same local network as your Home Assistant instance
- Powered on and broadcasting UDP packets on port 16121

{% include integrations/config_flow.md %}

When adding the integration, it will automatically listen for UDP broadcasts for approximately 10 seconds. If your EARN-E device is found, you will see a confirmation screen with its IP address.

If no device is discovered (for example, if the meter is on a different subnet), you will be asked to enter the IP address manually. The integration will then listen for data from that IP address to verify connectivity.

{% configuration_basic %}
IP Address:

>   description: "The local IP address of your EARN-E P1 Meter (e.g., 192.168.1.100). Only required if auto-discovery does not find your device."

Reviewer 2: ```suggestion
  description: "The local IP address of your EARN-E P1 Meter (for example, 192.168.1.100). Only required if auto-discovery does not find your device."
```

{% endconfiguration_basic %}

## Reconfiguration

If the IP address of your EARN-E device changes (for example, after a DHCP renewal), you can update it through the reconfigure option:

1. Go to {% my integrations title="**Settings** > **Devices & services**" %}.

> 2. On the **EARN-E P1 Meter** integration, select the three-dot menu.

Reviewer 2: ```suggestion
2. On the **EARN-E P1 Meter** integration, select the three dots {% icon "mdi:dots-vertical" %} menu.
```

3. Select **Reconfigure**.

> 4. Enter the new IP address.

Reviewer: There's no reconfiguration flow anymore

Author: Good catch — verified against the merged `config_flow.py`, there's no reconfigure step. Removed the whole section.

---
title: EARN-E P1 Meter
description: Monitor your smart meter's energy and gas data in real time with the EARN-E P1 Meter integration.
ha_category:
  - Energy
  - Sensor

> ha_release: "2026.4"

Reviewer 2: ```suggestion
ha_release: 2026.4
```
these " are not needed here, (except for .10)

ha_iot_class: Local Push
ha_config_flow: true
ha_codeowners:
  - '@Miggets7'
ha_domain: earn_e_p1
ha_platforms:
  - sensor
ha_integration_type: device
---

The **EARN-E P1 Meter** {% term integration %} connects to the [EARN-E energy monitor](https://earn-e.com/product/energiemonitor/), a device that reads your smart meter's P1 port and broadcasts real-time energy data via UDP on the local network. This integration listens for those broadcasts and exposes them as sensor entities.

No cloud connection or polling is needed — the device pushes data directly to Home Assistant over the local network.

## Supported devices

- [EARN-E energy monitor](https://earn-e.com/product/energiemonitor/)

## Supported functionality

> The following sensor entities are created:

Reviewer 2: ```suggestion
The **EARN-E P1 Meter** integration provides the following entities.
```
If you think there will be other entities supported in the future (in addition to sensors), I would keep this intro generic.

### Real-time sensors (~1 second updates)

> - **Power Delivered** — Current power being consumed (kW)

Reviewer 2: ```suggestion
- **Power delivered**: Current power being consumed (kW)
```
We use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization). Of course, this needs to match the actual UI string. If you keep heading style capitalization there for some reason, the string here needs to match.

Also applies to the other sensor names.

> - **Power Returned** — Current power being returned to the grid (kW)

Reviewer 2: ```suggestion
- **Power returned to grid**: Current power being returned to the grid (kW)
```
Just an idea. Needs to match the actual sensor name.

- **Voltage L1** — Voltage on phase 1 (V)

> - **Current L1** — Current on phase 1 (A)

Reviewer 2: ```suggestion
- **Voltage L1**: Voltage on phase 1 (V)
- **Current L1**: Current on phase 1 (A)
```

### Meter reading sensors (~60 second updates)

- **Energy Delivered Tariff 1** — Total energy consumed on tariff 1 (kWh)
- **Energy Delivered Tariff 2** — Total energy consumed on tariff 2 (kWh)
- **Energy Returned Tariff 1** — Total energy returned on tariff 1 (kWh)
- **Energy Returned Tariff 2** — Total energy returned on tariff 2 (kWh)
- **Gas Delivered** — Total gas consumed (m³)
- **Wi-Fi RSSI** — Wi-Fi signal strength of the device (dBm)

## Prerequisites

The EARN-E energy monitor must be:

- Connected to your smart meter's P1 port
- Connected to the same local network as your Home Assistant instance
- Powered on and broadcasting UDP packets on port 16121

> ## Configuration

Reviewer 2: ```suggestion
```
The title is not needed here. It is part of the include element.
